### PR TITLE
Include supplier info in ride

### PR DESCRIPTION
**Pull request by peteruithoven**
Includes the supplier info that was added in #17 in the ride page. 
Also lists the orders per supplier. 
Ditched Datatables for now as it couldn't easily handle sub tables that are not dynamically created.

### CODE REVIEW

**Comment by peteruithoven** [comment]
Example of end result:
Screenshot from 2019-03-16 11-46-36@2x

**Comment by rikva** [comment]
Looks good, I'll have a look at the Python code. What's the rationale behind showing the orders btw?

**Comment by peteruithoven** [comment]
Ah, it enables drivers to better guess how many crates to bring and more important in some cases the the supplier leaves it somewhere and isn't present so it's nice to be able to verify you've found everything.

**Comment by rikva** [comment]
The Python logic is copied from here, right?
https://github.com/rikva/voko/blob/5cdf71a4b0eeca0ea1413d4724919b6dfa76b3f6/webapp/ordering/admin_views.py#L42

I'm thinking we could move it to the OrderRound or Supplier model, this is very un-DRY ;)

**Comment by peteruithoven** [comment]
Good point. I'll try merging that. 

I also just released I didn't the suppliers on the Ride's route.

**Comment by rikva** [comment]
Unfortunately a few conflicts!

**Comment by peteruithoven** [comment]
I'll look at the merge conflicts, but a bigger issue is that I can't figure out how to filter the orders_per_supplier on the suppliers that exist on a Ride's Route. 

Simple example of what I'm trying to do:
``` python
# filtering dict on specific keys that might not exist
old_dict = {'a': 1, 'b': 2, 'c': 3}
wanted_keys = ['a', 'c', 'd']
new_dict = { key: old_dict[key] for key in wanted_keys if key in old_dict}
for key in new_dict.keys(): print (("key: %s" % key))
```

I've tried:
``` python
class Ride(TimeStampedModel):
    ...
    def orders_per_supplier(self):
        orders_per_supplier = self.order_round.orders_per_supplier
        suppliers_in_route = self.route.suppliers.all()
        supplier_names_in_route = [s.name for s in suppliers_in_route]

        return {
            key: orders_per_supplier[key]
            for key in supplier_names_in_route
            if key in orders_per_supplier
        }
```
I've also tried:
``` python
class Ride(TimeStampedModel):
    ...
    def orders_per_supplier(self):
        orders_per_supplier = self.order_round.orders_per_supplier
        suppliers_in_route = self.route.suppliers.all()
        supplier_names_in_route = [s.name for s in suppliers_in_route]

        filtered = {}
        for supplier in orders_per_supplier.keys():
            if supplier in supplier_names_in_route:
                filtered[supplier] = orders_per_supplier[supplier]
        return filtered
```
I've tried many many things. But it never seems to properly compare the supplier names for some reason. I'm kind of stuck. Do you have a clue what might be going on here?

**Comment by rikva** [comment]
I think what's going wrong is that `self.order_round.orders_per_supplier` returns a dict where the keys are `Supplier` instances, not just their names. So you don't need `supplier_names_in_route` at all.

This would probably work:
```
In [25]: {
    ...:         key: orders_per_supplier[key]
    ...:         for key in suppliers_in_route
    ...:         if key in orders_per_supplier
    ...: }

Out[25]: 
{<Supplier: Nieuw Slagmaat>: {'orderproducts': [], 'sum': Decimal('21.75')},
 <Supplier: De Kas Eet & Kweekplek>: {'orderproducts': [], 'sum': 0},
 <Supplier: Koningshof>: {'orderproducts': [], 'sum': 0},
 <Supplier: Geertjes Hoeve>: {'orderproducts': [], 'sum': 0}}

```

I cringe a little bit seeing all this code I wrote 5 years ago, these are operations that should be executed by the database instead of Python. 😭 

Having said that, this approach is fine, I think we both don't want to refactor code that works fine but could perform a little bit better.

**Comment by peteruithoven** [comment]
Ah! That makes sense. I was confused because when you log a Supplier it's name is simply returned. 

Yeah for 70 users performance in not a priority ;)

Now this one is finally really ready for review.

File: webapp/ordering/admin_views.py [hunk 1]
```diff
@@ -39,41 +39,6 @@ class OrderAdminOrderLists(GroupRequiredMixin, DetailView):
...
-
-                data[supplier]['orderproducts'].append(
-                    {'product': product,
-                     'amount': product_sum,
-                     'sub_total': product_sum * product.base_price}
-                )
-
-        return data
-
-    def _get_total_prices_per_supplier(self, supplier, order_round):
```
Comment (rikva) [inline webapp/ordering/admin_views.py]: Nice that you removed some redundancy here.

**Comment by rikva** [comment]
Merged 👍

**Pull request merged**